### PR TITLE
get /api/categories, testing, error handling, initial code setup

### PR DESCRIPTION
**Pull request by xViolaine**
I have added a test folder as well as a test file, in which I'm testing for the first get request (getting all categories) as well as a test to handle 404 errors.

I have set up the structure of my code in an MVC pattern, I've added a listening file to actually see my requests with Insomnia, I've set up my app.js and handled Internal Server Errors there as well!

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const express = require('express');
+const { getAllCategories } = require('./controllers/controller-games');
+
+const app = express();
+
+app.use(express.json());
```
Comment (daviidbartlett) [inline app.js:6]: why is this here?

**Review by daviidbartlett — APPROVED** [review]
A few bits of code which are not required. Be careful with blindly following what you've done previously.

File: app.js (hunk 1 above)
Comment (xViolaine) [reply app.js:6]: Well, further down the line we are going to do POST and PATCH requests, so this is a method to handle that better.
I do agree that I should definitely JUST try to add the things specified in the ticket, especially when thinking about how actually coding in a team would go -> could cause merging conflicts etc.

I'll ensure to not get too far ahead of myself and instead just do the things one by one, ticket by ticket - thank you!

**Comment by xViolaine** [comment]
> A few bits of code which are not required. Be careful with blindly following what you've done previously.

Ah, but I didn't - I just added the listen file cause I wanted to see what it would look like using Insomnia, but you are right: If I keep building up my test library carefully, then this will be all the proof I need that my stuff is "working".

Thank you for your advice and constructive criticism! :)
